src/llfs/storage_file_builder.test.cpp

```diff
@@ -241,4 +358,97 @@ TEST_F(StorageFileBuilderTest, WriteReadFile)
   ASSERT_TRUE(recovered_device.ok()) << BATT_INSPECT(recovered_device.status());
 }
 
+//---------------------------------------------------------------------------------------------------
+
+TEST_F(StorageFileBuilderTest, WriteReadManyPackedConfigs)
+{
+  i64 base_file_offset = 0;
+  llfs::StatusOr<llfs::ScopedIoRing> ioring =
+      llfs::ScopedIoRing::make_new(llfs::MaxQueueDepth{1024}, llfs::ThreadPoolSize{1});
+
+  ASSERT_TRUE(ioring.ok()) << BATT_INSPECT(ioring.status());
+
+  const char* const test_file_name = "/tmp/llfs_test_file";
+  std::filesystem::remove(test_file_name);
+  boost::uuids::uuid page_device_uuid;
+
+  llfs::StatusOr<int> test_fd =
+      llfs::create_file_read_write(test_file_name, llfs::OpenForAppend{false});
+  ASSERT_TRUE(test_fd.ok()) << BATT_INSPECT(test_fd.status());
+  {
+    llfs::Status status = llfs::enable_raw_io_fd(*test_fd, /*enabled=*/true);
```

Comment:
There's a subtle resource management issue here: if the call to `enable_raw_io_fd` fails, then the fd is never closed (it is closed in the success path because ownership is transferred to `IoRingRawBlockFile test_file` below).  The easiest fix I think is to change this `ASSERT_TRUE` to an `EXPECT_TRUE`.

Reply:
Good catch!  Thanks for fixing both.